docs/reference/polykey-core/identities.md

```diff
@@ -1,7 +1,286 @@
-# Identification with Third Party Services
+# Identities
 
-TBD, must take information from reference and add it here as a complete spec of Social Identity Discovery.
+Identities are Polykey's method of establishing a gestalt which enables social discovery and sharing of vaults with ease. This incorporates an inherit trust in the
+identity and the identity provider. This enables vaults to be securely cloned between identities without having to rely on nodeIDs.
 
-References:
+## Identity
 
-- https://gitlab.com/MatrixAI/Engineering/Polykey/polykey-design/-/issues/31
+Every individual identity is composed of two elements
+
+1. A provider ID, which is an identity provider akin to github.com, or tentatively in the future, meta, or X.
+2. An Identity ID, which without the domain of a provider is an uniquely identifiable identity. This is usually a username on the providers' service.
+
+```ts
+type IdentityData = {
+  providerId: ProviderId;
+  identityId: IdentityId;
+  name?: string;
+  email?: string;
+  url?: string;
+};
+```
+
+## Specification
+
+### `IdentitiesManager`
+
+The `IdentitiesManager` class is repsonible for handling Gestalt IDs and linking them with nodeIDs.
+It maps ProviderIDs and associated IdentityIDs together and into a gestalt.
+
+---
+
+#### `type ProviderId`
+
+Identifies an identity provider
+
+---
+
+#### `type IdentityId`
+
+Unique identifier for an identity within a providers domain. Usually in form of an username.
+
+---
+
+#### `type ProviderToken`
+
+Authentication tokens from the providers' API, used to authenticate Identity IDs.
+
+---
+
+#### `type ProviderTokens`
+
+Authentication tokens indexed by IdentityIDs
+
+---
+
+#### `createIdentitiesManager(...)`
+
+Takes an object with the following properties:
+
+- db
+  - sigchain
+  - keyRing
+  - gestaltGraph
+  - logger
+
+Constructs an instance of IdentitiesManager.
+The identities manager needs to be started with .start()
+
+---
+
+#### `public async start(): void`
+
+Initializes and starts the identity manager. This method is responsible for setting up necessary configurations and initializing any required connections or services. It must be called before using the identity manager to ensure that all components are properly initialized.
+
+#### Usage Example:
+
+```typescript
+await identitiesManager.start();
+// The identity manager is now ready to be used
+```
```

Comment:
So for example I don't think we should be done individual method docs here at all. It's too brittle.

Reply:
I agree, I noticed the same when i was copying over from `feature-launch-revamp`

Reply:
We will resolve in subsequent issues.